### PR TITLE
cache max size metric collector

### PR DESCRIPTION
**Pull request by khushbr**
*Issue #, if available:* N/A

*Description of changes:* Adding code change for collecting cache max size metrics. Unlike Cache Hit, Miss, Eviction and Cache Size which are collected on per shard basis, the cache max size is a node level static setting and should be collected through a separate collector, 'CacheCustomMetricsCollector' here(Open to renaming). 

```
sqlite> select * from Cache_MaxSize;
FIELD_DATA_CACHE|-1.0|-1.0|-1.0|-1.0
SHARD_REQUEST_CACHE|5194383.0|5194383.0|5194383.0|5194383.0
sqlite>
```
Pending UTs, I will add them as part of next iteration once we have the collector name and other variables(Cache Table Name, Dimension and Value) finalized. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        // (depending on whether the patch has been applied or not). Thus, we need to use reflection
+        // to check whether getMaxWeight() method exist in Cache.java
+        //
+        // Currently, we are collecting maxWeight metrics only for FieldData and Shard Request Cache.
+        CacheMaxSizeStatus fieldDataCacheMaxSizeStatus = AccessController.doPrivileged(
+                (PrivilegedAction<CacheMaxSizeStatus>) () -> {
+                    try {
+                        Cache fieldDataCache = indicesService.getIndicesFieldDataCache().getCache();
+                        long fieldDataMaxSize = (Long) FieldUtils.readField(fieldDataCache, "maximumWeight", true);
+                        return new CacheMaxSizeStatus("Field Data Cache", fieldDataMaxSize);
```
Comment (vigyasharma) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java:73]: Should the cache name here (`"Field Data Cache"`) come from a declared String constant or a protobuf resource enum? Same for `"Shard Request Cache"`

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        // This is for backward compatibility. Core ES may or may not emit custom shard metrics
+        // (depending on whether the patch has been applied or not). Thus, we need to use reflection
+        // to check whether getMaxWeight() method exist in Cache.java
+        //
+        // Currently, we are collecting maxWeight metrics only for FieldData and Shard Request Cache.
+        CacheMaxSizeStatus fieldDataCacheMaxSizeStatus = AccessController.doPrivileged(
+                (PrivilegedAction<CacheMaxSizeStatus>) () -> {
+                    try {
+                        Cache fieldDataCache = indicesService.getIndicesFieldDataCache().getCache();
+                        long fieldDataMaxSize = (Long) FieldUtils.readField(fieldDataCache, "maximumWeight", true);
```
Comment (rguo-aws) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java:72]: should we create a dedicated Const.java and define the string constants used by PerformanceController in it ?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java (hunk 1 above)
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java:73]: +1

Comment (khushbr) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java]: I have moved them to an enum in the AllMetrics.java in the reader. Given we are using this file for defining metric level enums and we have a dependency on reader jar.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java (hunk 2 above)
Comment (khushbr) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/CacheCustomMetricsCollector.java]: Moved them to a const file under Decision maker.

**Pull request merged**